components/style/data.rs

```diff
@@ -39,14 +40,18 @@ pub struct NodeStyles {
     /// The results of CSS styling for this node.
     pub primary: Arc<ComputedValues>,
 
+    /// The rule node representing the last rule matched for this node.
+    pub rule_node: StrongRuleNode,
```

Comment:
Can you group these two into:

pub struct StyleResult {
    pub rule_node: Arc<ServoRuleNode>,
    pub computed_values: Arc<ServoComputedValues>,
}

That will match the algorithm at [1], and allow us to make PseudoStylesInner a map between PseudoElement and StyleResult. This will also allow us to transfer ownership of both to the frame.

[1] https://docs.google.com/document/d/1TcTAQMm-jIiNgzkDgpO04Pja_aQNRNkrD0CNCCYfbo8